integration/bug-report-test.ts

```diff
@@ -58,30 +58,34 @@ test.beforeAll(async () => {
     // `createFixture` will make an app and run your tests against it.
     ////////////////////////////////////////////////////////////////////////////
     files: {
+      "app/utils/buggy.server.ts": js`
+        export function buggy() {
+          return "this is a bug";
+        }
+      `,
+      "app/routes/about.tsx": js`
+        import { buggy } from "../utils/buggy.server";
+        const forcebug = buggy();
```

Comment:
This is your issue - components need to render both on the server (initial document request) and on the client (initial page hydration and subsequent client-side navigations).  So you cannot user server-only modules in components if you intend to hydrate your app into a SPA and do client-side navigations via `<Link>`.  Server-only modules can only be used in code that _only_ runs on the server - such as loaders and actions.

If you _want_ to use server only-modules in your components, you can do so if you treat your app as a MPA and do not include the `<Scripts>` components - this preventing Remix from hydrating your app into a SPA.